analyzers/src/SonarAnalyzer.CSharp/Metrics/CSharpExecutableLinesMetric.cs

```diff
@@ -112,65 +108,40 @@ private bool FindExecutableLines(SyntaxNode node)
                     case SyntaxKind.StructDeclaration:
                     case SyntaxKind.ClassDeclaration:
                     case SyntaxKindEx.RecordClassDeclaration:
-                        return !HasExcludedCodeAttribute((BaseTypeDeclarationSyntax)node, btdc => btdc.AttributeLists, canBePartial: true);
+                    case SyntaxKindEx.RecordStructDeclaration:
+                        return !HasExcludedCodeAttribute((BaseTypeDeclarationSyntax)node, x => x.AttributeLists, true);
 
                     case SyntaxKind.MethodDeclaration:
                     case SyntaxKind.ConstructorDeclaration:
-                        return !HasExcludedCodeAttribute((BaseMethodDeclarationSyntax)node, bmds => bmds.AttributeLists, canBePartial: true);
+                        return !HasExcludedCodeAttribute((BaseMethodDeclarationSyntax)node, x => x.AttributeLists, true);
 
                     case SyntaxKind.PropertyDeclaration:
                     case SyntaxKind.EventDeclaration:
-                        return !HasExcludedCodeAttribute((BasePropertyDeclarationSyntax)node, bpds => bpds.AttributeLists);
+                        return !HasExcludedCodeAttribute((BasePropertyDeclarationSyntax)node, x => x.AttributeLists, false);
 
                     case SyntaxKind.AddAccessorDeclaration:
                     case SyntaxKind.RemoveAccessorDeclaration:
                     case SyntaxKind.SetAccessorDeclaration:
                     case SyntaxKind.GetAccessorDeclaration:
                     case SyntaxKindEx.InitAccessorDeclaration:
-                        return !HasExcludedCodeAttribute((AccessorDeclarationSyntax)node, ads => ads.AttributeLists);
+                        return !HasExcludedCodeAttribute((AccessorDeclarationSyntax)node, ads => ads.AttributeLists, false);
 
                     default:
                         return true;
                 }
             }
 
-            private bool HasExcludedCodeAttribute<T>(T node,
-                                                     Func<T, SyntaxList<AttributeListSyntax>> getAttributeLists,
-                                                     bool canBePartial = false)
-                where T : SyntaxNode
+            private bool HasExcludedCodeAttribute<T>(T node, Func<T, SyntaxList<AttributeListSyntax>> getAttributeLists, bool canBePartial) where T : SyntaxNode
```

Comment:
Removed optional, because it was 2x explicit and 2x implicit => could be just explicit everywhere